x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/TransactionPropertiesTableForFlyout.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignore
+import { EuiTab } from '@elastic/eui';
+import { capitalize, first, get } from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { SERVICE_AGENT_NAME } from '../../../../../common/constants';
+import { Transaction } from '../../../../../typings/Transaction';
+import { IUrlParams } from '../../../../store/urlParams';
+// @ts-ignore
+import { fromQuery, history, toQuery } from '../../../../utils/url';
+import {
+  getPropertyTabNames,
+  PropertiesTable
+} from '../../../shared/PropertiesTable';
+// @ts-ignore
+import { Tab } from '../../../shared/UIComponents';
+
+// Ensure the selected tab exists or use the first
+function getCurrentTab(tabs: string[] = [], selectedTab: string | null) {
+  return selectedTab && tabs.includes(selectedTab) ? selectedTab : first(tabs);
+}
+
+function getTabs(transactionData: Transaction) {
+  const dynamicProps = Object.keys(transactionData.context);
+  return getPropertyTabNames(dynamicProps);
+}
+
+interface Props {
+  location: Location;
+  transaction: Transaction;
+  urlParams: IUrlParams;
+}
+
+export const TransactionPropertiesTableForFlyout: React.SFC<Props> = ({
+  location,
+  transaction,
+  urlParams
+}) => {
+  const tabs = getTabs(transaction);
+  const currentTab = getCurrentTab(tabs, urlParams.flyoutDetailTab);
+  const agentName = get(transaction, SERVICE_AGENT_NAME, 'n/a');
```

Comment:
The default value is never used, so why set it?

Reply:
it'll try to find docs for an agent called `n/a` which (hopefully) doesn't return anything.

Reply:
Because typescript complained without it because it always expects a string. Need to find where that is typed that way and make it `?` optional probably.

Reply:
Note: had to make a few changes to documentation.ts, getFeatureDocs and translateAgentName needed to account for an undefined agentName. Good catch.